src/expandable-section/expandable-section-header.tsx

```diff
@@ -190,12 +188,19 @@ const ExpandableHeaderTextWrapper = ({
         </InternalHeader>
       ) : (
         <>
-          <HeadingTag
-            className={clsx(styles['header-wrapper'], headingTagListeners && styles['click-target'])}
-            {...headingTagListeners}
-          >
-            {headerButton}
-          </HeadingTag>
+          {variant === 'compact' ? (
+            <div className={clsx(styles['header-wrapper'], styles['header-wrapper-compact'])}>
+              <HeadingTag className={styles['click-target']}>{headerButton}</HeadingTag>
+              {headerActions}
```

Comment:
I have refactored this and now the condition does not depend on whether the variant is `compact` but on whether actions are present and supported for the variant. This also brings us closer to support actions for default and footer variants (AWSUI-21326) and when that happens this can be unified.

Although the slot is called `headerActions`, I think using it for other elements is OK, and this is less confusing or forced than other options I can think of. Note that `headerText` is a string-only prop and the ReactNode `header` prop is deprecated.